internal/stats/query.go

```diff
@@ -31,6 +33,41 @@ type BaseQuery struct {
 	// Define how the results should be sorted by time.
 	// -1 for ascending, 0 for no sorting, 1 for descending.
 	Sort int `json:"sort"`
+
+	// Filter results to only include metrics that match BaseMetric attribute field values.
+	// Keys are field names (e.g., "course") and values are what to include (e.g., course101).
+	// This filter is applied after all other BaseQuery conditions are applied.
+	Where map[string]any `json:"where,omitempty"`
+}
+
+type MetricQuery struct {
+	BaseQuery
```

Comment:
Do we even need a base query anymore?
Now that the metrics are all the same, can we just have query?